src/EditorFeatures/CSharpTest/AssignOutParameters/AssignOutParametersTests.cs

```diff
@@ -0,0 +1,534 @@
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForExpressionBodyMember()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i) => [|'a';|]
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        return 'a';
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForLambdaExpressionBody()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) => [|'a';|]
+    }
+}",
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) => { i = 0; return 'a'; };
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingForLocalFunctionExpressionBody()
+        {
+            // Note desirable.  Happens because error is not reported on expr-body
+            // like for lambdas.
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void X()
+    {
+        char D(out int i) => [|'a';|]
+        D(out _);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForLambdaBlockBody()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) =>
+        {
+            [|return 'a';|]
+        }
+    }
+}",
+@"class C
+{
+    delegate char D(out int i);
+    void X()
+    {
+        D d = (out int i) =>
+        {
+            i = 0;
+            return 'a';
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForLocalFunctionBlockBody()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void X()
+    {
+        char D(out int i)
+        {
+            [|return 'a';|]
+        }
+
+        D(out _);
+    }
+}",
+@"class C
+{
+    void X()
+    {
+        char D(out int i)
+        {
+            i = 0;
+            return 'a';
+        }
+
+        D(out _);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestFixAll1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(bool b, out int i, out int j)
+    {
+        if (b)
+        {
+            {|FixAllInDocument:return 'a';|}
+        }
+        else
+        {
+            return 'a';
+        }
+    }
+}",
+@"class C
+{
+    char M(bool b, out int i, out int j)
+    {
+        if (b)
+        {
+            i = 0;
+            j = 0;
+            return 'a';
+        }
+        else
+        {
+            i = 0;
```

Comment:
Another case where the user likely wants initialization at start of the method so all the exit points where out params are not initialized are handled. Maybe a fix should also appear at the _parameter definition node_ that adds initialization at start of the method?

Reply:
> Maybe a fix should also appear at the parameter definition node that adds initialization at start of the method?

having the fix be at a different location than the squiggles semes confusing.  But i do agree that having a way to just pre-initialize at the start makes a lot of sense.

Reply:
That is reasonable. Maybe that should just be a separate refactoring, which need not be part of this PR.

Reply:
. That is reasonable. Maybe that should just be a separate refactoring, which need not be part of this PR.

Agreed :)